languages/tree-sitter-stack-graphs-java/src/stack-graphs.tsg

```diff
@@ -355,46 +866,136 @@ attribute node_symbol = node            => symbol = (source-text node), source_n
   edge expr_ref__typeof -> expr_ref
 }
 
-(local_variable_declaration
-  type: (type_identifier) @type
-  declarator: (variable_declarator
-    name: (identifier) @identifier
-    value: (object_creation_expression
-      type: (type_identifier) @object)
-  )) @local_variable {
-  node @object.value
-  attr (@object.value) node_reference = @object
-  node @type.value
-  attr(@type.value) node_reference = @type
-  edge @type.value -> @local_variable.after_scope
-  edge @local_variable.after_scope -> @object.value
-  edge @object.value -> @local_variable.after_scope
+(method_reference . (_) @lhs) @this {
+  edge @lhs.lexical_scope -> @this.lexical_scope
+}
+
+(method_reference . (_) @lhs (identifier) @id) @this {
+  node @this.member
+
+  node expr_ref
+  attr (expr_ref) node_reference = @id
+  edge expr_ref -> @this.member
+
+  attr (@this.member) push_symbol = "."
+  edge @this.member -> @lhs.type
+
+  node expr_ref__typeof
+  ; edge @this.type -> expr_ref__typeof ; FIXME: function types
+
+  attr (expr_ref__typeof) push_symbol = ":"
+  edge expr_ref__typeof -> expr_ref
+}
+
+(method_reference . (identifier) @lhs (identifier) @id) @this {
+  ; @lhs could be a type name
+  node ref
+  attr (ref) node_reference = @lhs
+  edge @this.member -> ref
+  edge ref -> @this.lexical_scope
+}
+
+(parenthesized_expression (_) @child) @expr {
+  edge @child.lexical_scope -> @expr.lexical_scope
 }
 
 ;; =============
 ;;  Expressions
 ;; =============
 
 [
-  ; FIXME: all expressions should be here
-  (assignment_expression)
+  (array_initializer)
+  (expression)
   (field_access)
-  (identifier)
-  (method_invocation)
+  (parenthesized_expression)
+  (primary_expression)
   (super)
-  (this)
 ] @expr
 {
   node @expr.type
   node @expr.lexical_scope
 }
 
+(assignment_expression left: (identifier) @ident) @this {
+  ; FIXME: can we combine this with primary_expression/identifier?
```

Comment:
Look I know lvalues and rvalues aren't the same thing semantically at all, but argh, seriously, state kills.